src/components/TextField/TextField.tsx

```diff
@@ -20,14 +20,33 @@ export const TextField: React.FC<Props> = ({
   label = name,
   placeholder = `Enter ${label}`,
   required = false,
-  onChange = () => {},
+  onChange,
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
-
-  // To show errors only if the field was touched (onBlur)
+  const [fieldValue, setFieldValue] = useState(value);
   const [touched, setTouched] = useState(false);
-  const hasError = touched && required && !value;
+  const [hasUrlError, setHasUrlError] = useState(false);
+
+  const hasError = (touched && required && !fieldValue) || hasUrlError;
```

Comment:
```suggestion
  const hasError = (touched && required && !fieldValue.trim()) || hasUrlError;
```
now no any error if type only spaces in input. trim fieldValue for show error if in input all characters are spaces.